src/arithmetic.s

```diff
@@ -0,0 +1,1311 @@
+.file	"arithmetic.s"
+.text
+
+# void bls12_381::_add(fp* z, const fp* x, const fp* y)
+.globl _ZN9bls12_3814_addEPNS_2fpEPKS0_S3_
```

Comment:
I don't think using mangled C++ names here is a typical pattern -- maybe even considered a bad pattern -- but would require more refactoring to move them out of the bls namespace to be plain ole nonmangled names

Reply:
What are those names? `llvm-cxxfilt` or `c++filt` failed to demangle them in my try.

Reply:
> What are those names? `llvm-cxxfilt` or `c++filt` failed to demangle them in my try.

Strange, works for me:
```
❯ c++filt _ZN9bls12_3814_addEPNS_2fpEPKS0_S3_
bls12_381::_add(bls12_381::fp*, bls12_381::fp const*, bls12_381::fp const*)
 ❯ llvm-cxxfilt-19 _ZN9bls12_3814_addEPNS_2fpEPKS0_S3_
bls12_381::_add(bls12_381::fp*, bls12_381::fp const*, bls12_381::fp const*)
```